protocols/models/models.go

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package models
+
+import (
+	"fmt"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type MetricsModelTranslator interface {
+	// MetricsFromModel converts a data model of another protocol into pdata.
+	MetricsFromModel(src interface{}) (pdata.Metrics, error)
```

Comment:
I think for consistency we either split these into 2 interfaces or merge the Encoder/Decoder to be consistent with this.